prow/cmd/deck/main.go

```diff
@@ -301,8 +307,50 @@ func main() {
 	// signal to the world that we're ready
 	health.ServeReady()
 
+	// cookie secret will be used for CSRF protection and should be exactly 32 bytes
+	// we sometimes accept different lengths to stay backwards compatible
+	var csrfToken []byte
+	if o.cookieSecretFile != "" {
+		cookieSecretRaw, err := loadToken(o.cookieSecretFile)
+		if err != nil {
+			logrus.WithError(err).Fatal("Could not read cookie secret file")
+		}
+		decodedSecret, err := base64.StdEncoding.DecodeString(string(cookieSecretRaw))
+		if err != nil {
+			logrus.WithError(err).Fatal("Error decoding cookie secret")
+		}
+		if len(decodedSecret) == 32 {
+			csrfToken = decodedSecret
+		}
+		if len(decodedSecret) > 32 {
+			logrus.Warning("Cookie secret should be exactly 32 bytes. Consider truncating the existing cookie to that length")
+			hash := sha256.Sum256(decodedSecret)
+			csrfToken = hash[:]
+		}
+		if len(decodedSecret) < 32 {
+			if o.rerunCreatesJob {
+				logrus.Fatal("Cookie secret must be exactly 32 bytes")
+				return
+			}
+			logrus.Warning("Cookie secret should be exactly 32 bytes")
+		}
+	}
+
+	if o.rerunCreatesJob && csrfToken == nil {
+		logrus.Warning("Rerun creates job cannot be enabled without CSRF protection, which requires --cookie-secret to be exactly 32 bytes")
+		o.rerunCreatesJob = false
+	}
+
+	// if we allow direct reruns, we must protect against CSRF in all post requests using the cookie secret as a token
+	// for more information about CSRF, see https://github.com/kubernetes/test-infra/blob/master/prow/cmd/deck/csrf.md
+	if o.rerunCreatesJob || csrfToken != nil {
```

Comment:
This logic is confusing — if `csrfToken` is nil, this will fail anyway. Additionally, if `csrfToken == nil` then `rerunCreatesJob` is forced to `false` by the block above.

Simplify this to just `csrfToken != nil` and move comments relating to reruns to the above block